validation/open_boundaries/cylinder.jl

```diff
@@ -1,127 +1,214 @@
-# This validation script shows open boundaries working in a simple case where the
-# flow remains largely unidirectional and so at one end we have no matching scheme
-# but just prescribe the inflow. At the other end we then make no assumptions about
-# the flow and use a very simple open boundary condition to permit information to 
-# exit the domain. If, for example, the flow at the prescribed boundary was reversed
-# then the model would likely fail.
+using Oceananigans
+using Oceananigans.Models.NonhydrostaticModels: ConjugateGradientPoissonSolver
+using Oceananigans.Solvers: DiagonallyDominantPreconditioner
+using Oceananigans.Operators: ℑxyᶠᶜᵃ, ℑxyᶜᶠᵃ
+using Oceananigans.Solvers: FFTBasedPoissonSolver
+using Printf
+using CUDA
 
-using Oceananigans, CairoMakie
-using Oceananigans.BoundaryConditions: FlatExtrapolationOpenBoundaryCondition
+using Oceananigans.BoundaryConditions: PerturbationAdvectionOpenBoundaryCondition
 
+u∞ = 1
+r = 1/2
+arch = GPU()
+stop_time = 200
 
-@kwdef struct Cylinder{FT}
-    D :: FT = 1.0
-   x₀ :: FT = 0.0
-   y₀ :: FT = 0.0
-end
+cylinder(x, y) = (x^2 + y^2) ≤ r^2
 
-@inline (cylinder::Cylinder)(x, y) = ifelse((x - cylinder.x₀)^2 + (y - cylinder.y₀)^2 < (cylinder.D/2)^2, 1, 0)
+"""
+    drag(modell; bounding_box = (-1, 3, -2, 2), ν = 1e-3)
 
-architecture = GPU()
+Returns the drag within the `bounding_box` computed by:
 
-# model parameters
-Re = 200
-U = 1
-D = 1.
-resolution = D / 40
+```math
```

Comment:
Actually the pressure contribution shouldn't be included here since ``G^n`` doesn't include it right?

Reply:
I can't see the line you're referring to